docs/events.md

```diff
@@ -37,6 +37,33 @@ registered startup handlers have completed.
 The shutdown handlers will run once all connections have been closed, and
 any in-process background tasks have completed.
 
+A single lifespan asynccontextmanager handler can be used instead of
+separate startup and shutdown handlers:
+
+```python
+import contextlib
+import anyio
+from starlette.applications import Starlette
+
+
+@contextlib.asynccontextmanager
+async def lifespan(app):
+    # acquire async resources
+    async with anyio.create_task_group() as app.tg:
+        try:
+            yield
+        finally:
+            with anyio.CancelScope(shield=True):
+                # release async resources
```

Comment:
sorry I lead you wrong here - the TaskGroup context manager will capture any CancelledError for you

```suggestion
    async with anyio.create_task_group() as app.tg:
        yield
    # release async resources
```

Reply:
I do want to somehow document that `yield` can now throw CancelledError

Reply:
sometimes you want to acquire/release resources inside the TaskGroup and sometimes you want to acquire/release them outside the TaskGroup and there's no specific answer as they both have different behaviours with their own pros/cons

![image](https://user-images.githubusercontent.com/413772/124912492-ee183200-dfe5-11eb-8efd-41cb3e9708d8.png)

Reply:
I'm not sure we should have the main documentation for lifespan be setting `app.tg`.  That could be a separate topic, but I think the initial introduction to "make a lifespan" example should be as simple as possible.

```python
@contextlib.asynccontextmanager
async def lifespan(app):
    async with some_async_resource():
        yield
```

Reply:
that's fair

Reply:
I like the simple lifespan example; we can also add a link to [anyio documentation](https://anyio.readthedocs.io/) on task groups and cancellation.